packages/inference/src/HfInference.ts

```diff
@@ -21,22 +21,83 @@ type TaskWithNoAccessTokenNoEndpointUrl = {
 	) => ReturnType<Task[key]>;
 };
 
-export class HfInference {
-	private readonly accessToken: string;
-	private readonly defaultOptions: Options;
+export class HfInference implements Task {
+	protected readonly accessToken: string;
+	protected readonly defaultOptions: Options;
+
+	speechToText: typeof tasks.speechToText;
+	audioClassification: typeof tasks.audioClassification;
+	automaticSpeechRecognition: typeof tasks.automaticSpeechRecognition;
+	textToSpeech: typeof tasks.textToSpeech;
+	audioToAudio: typeof tasks.audioToAudio;
+	imageClassification: typeof tasks.imageClassification;
+	imageSegmentation: typeof tasks.imageSegmentation;
+	imageToText: typeof tasks.imageToText;
+	objectDetection: typeof tasks.objectDetection;
+	textToImage: typeof tasks.textToImage;
+	imageToImage: typeof tasks.imageToImage;
+	zeroShotImageClassification: typeof tasks.zeroShotImageClassification;
+	featureExtraction: typeof tasks.featureExtraction;
+	fillMask: typeof tasks.fillMask;
+	questionAnswering: typeof tasks.questionAnswering;
+	sentenceSimilarity: typeof tasks.sentenceSimilarity;
+	summarization: typeof tasks.summarization;
+	tableQuestionAnswering: typeof tasks.tableQuestionAnswering;
+	textClassification: typeof tasks.textClassification;
+	textGeneration: typeof tasks.textGeneration;
+	tokenClassification: typeof tasks.tokenClassification;
+	translation: typeof tasks.translation;
+	zeroShotClassification: typeof tasks.zeroShotClassification;
+	chatCompletion: typeof tasks.chatCompletion;
+	documentQuestionAnswering: typeof tasks.documentQuestionAnswering;
+	visualQuestionAnswering: typeof tasks.visualQuestionAnswering;
+	tabularRegression: typeof tasks.tabularRegression;
+	tabularClassification: typeof tasks.tabularClassification;
+	textGenerationStream: typeof tasks.textGenerationStream;
+	chatCompletionStream: typeof tasks.chatCompletionStream;
+
+	static mapInferenceFn<TOut, TArgs>(instance: HfInference, func: (...args: [TArgs, Options?]) => TOut) {
+		return function (...[args, options]: Parameters<(...args: [TArgs, Options?]) => TOut>): TOut {
+			return func({ ...args, accessToken: instance.accessToken }, { ...instance.defaultOptions, ...(options ?? {}) });
+		};
+	}
```

Comment:
This helper functions adds some default arguments when calling `HfInference.<task>()` while keeping the exact type for inputs and outputs